presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -158,28 +162,54 @@ private void validateShowStatsSubquery(ShowStats node, Query query, QuerySpecifi
             check(!querySpecification.getHaving().isPresent(), node, "HAVING is not supported in SHOW STATS SELECT clause");
             check(!querySpecification.getGroupBy().isPresent(), node, "GROUP BY is not supported in SHOW STATS SELECT clause");
             check(!querySpecification.getSelect().isDistinct(), node, "DISTINCT is not supported by SHOW STATS SELECT clause");
-
-            List<SelectItem> selectItems = querySpecification.getSelect().getSelectItems();
-            check(selectItems.size() == 1 && selectItems.get(0) instanceof AllColumns, node, "Only SELECT * is supported in SHOW STATS SELECT clause");
         }
 
-        private Node rewriteShowStats(ShowStats node, Table table, Constraint constraint)
+        private Node rewriteShowStats(ShowStats node, Table table, List<SelectItem> selectItems, Constraint constraint)
         {
+            List<String> statsColumnNames = buildColumnsNames();
             TableHandle tableHandle = getTableHandle(node, table.getName());
             TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, constraint);
-            List<String> statsColumnNames = buildColumnsNames();
-            List<SelectItem> selectItems = buildSelectItems(statsColumnNames);
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
+
             Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
-            List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, tableStatistics);
+            Set<String> extractNamedResultColumns = extractStatsColumns(tableMetadata, selectItems);
+            List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, extractNamedResultColumns, tableStatistics);
 
-            return simpleQuery(selectAll(selectItems),
+            return simpleQuery(selectAll(buildSelectItems(statsColumnNames)),
                     aliased(new Values(resultRows),
                             "table_stats_for_" + table.getName(),
                             statsColumnNames));
         }
 
-        private static void check(boolean condition, ShowStats node, String message)
+        private Set<String> extractStatsColumns(TableMetadata metadata, List<SelectItem> selectItems)
+        {
+            ImmutableSet.Builder<String> columns = ImmutableSet.builder();
+
+            for (SelectItem item : selectItems) {
+                if (item instanceof AllColumns) {
+                    for (ColumnMetadata column : metadata.getColumns()) {
+                        if (!column.isHidden()) {
+                            columns.add(column.getName());
+                        }
+                    }
+                }
```

Comment:
Add a test with `SHOW STATS FOR (SELECT *, nationkey ...`

Add a test with `SHOW STATS FOR (SELECT *, * ...`

Reply:
What is expected behaviour?

Reply:
Your impl won't produce duplicates and that's fine. 
It still should  not fail, so test is a good thing to have. 
Also, it will help us be conscious if we change the behavior later.

Reply:
`SHOW STATS FOR (SELECT *, nationkey ` I guess it should be coherent with regular `SELECT *, nationkey`, so I think we should  produce duplicate columns if user requested so. However I see no reason why he might do so.

In future we might want to support stats for arbitrary queries then that would like:
`SHOW STATS FOR (SELECT a+1, a+1, a+2)`. Then removing duplicates might be "surprising" to user.

Reply:
I guess this comment is not addressed.

Reply:
@findepi this tests are already there